### ISSUE TITLE
Error while training model that elasticsearch index data 

### ISSUE THREAD

**Issue by kang589** [regel/loudml#109]
Hi @regel. 
I have a elasticsearch problem with loudml-1.4.3. My model is below.

```
{
  "bucket_interval": "10m",
  "default_datasource": "weblog",
  "features": [
    {
      "default": 0,
      "field": "var_len",
      "metric": "max",
      "name": "test"
    }
  ],
  "interval": 60,
  "max_evals": 10,
  "name": "varlen-model",
  "offset": 30,
  "span": 5,
  "max_threshold": 0,
  "min_threshold": 0,
  "type": "timeseries"
}
```

But training is not working.

```
[root@Centos7 ~]# loudml create-model /etc/loudml/model.yml
/usr/lib64/python3.4/importlib/_bootstrap.py:321: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  return f(*args, **kwds)
INFO:root:model 'varlen-model' created
[root@Centos7 ~]# loudml train --from "2011-01-10T00:00:00Z" --to "2011-01-11T00:00:00Z" varlen-model
/usr/lib64/python3.4/importlib/_bootstrap.py:321: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  return f(*args, **kwds)
INFO:root:train(varlen-model) range=2011-01-10T00:00:00.000Z-2011-01-11T00:00:00.000Z train_size=0.670000 batch_size=64 epochs=100)
INFO:root:connecting to elasticsearch on 192.168.31.247:9200
ERROR:root:RequestError(400, 'search_phase_execution_exception', 'Fielddata is disabled on text fields by default. Set fielddata=true on [timestamp] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead.')
```

What exact means this error? 
Data type of 'var_len' field is number.
Data type of 'timestamp' field is date.
what should I do?

And i want to use loudml-elastic-1.5. Thanks.:)

**Comment by regel** [comment]
Hi @kang589 

The most common source of issues with Elasticsearch is index mapping. The ES API changes in each release (5.x, 6.x, and 7.x). The official Python libs made available by Elastic only support one major release.

The error that you see is caused by an index mapping that wasn't installed properly, so timestamp and keywords are not using the correct data types. You can run a curl query on the indices to get the actual index mapping and data types.

What is your ES version?

**Comment by regel** [comment]
Waiting for pull-request [145](https://github.com/regel/loudml/pull/145) to be merged, and that provides significant enhancement in Elasticsearch index mapping handling, compatible with all ES versions up to 7.x

**Comment by regel** [comment]
Solved on master (future 1.6.0 release)
